app/views/components/_autocomplete.html.erb

```
@@ -0,0 +1,43 @@
+<%
+  id ||= "autocomplete-#{SecureRandom.hex(4)}"
+  name ||= "autocomplete"
+  label ||= nil
+  value ||= nil
+  options ||= []
+
+  error_id = "error-#{SecureRandom.hex(4)}"
+  error_items ||= []
+
+  classes = %w(app-c-autocomplete govuk-form-group)
+  classes << "govuk-form-group--error" if error_items.any?
+
+  if !options.any? || !label
+    raise ArgumentError, "The autocomplete component needs at least one option and a label in order to render."
+  end
```

This looks good 👍  much safer than failing silently